.github/workflows/codeql-analysis.yml

```
@@ -39,6 +41,22 @@ jobs:
     steps:
     - name: Checkout repository
       uses: actions/checkout@v3
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.9.13
+        cache: 'pip'
+    - name: Upgrade pip, install poetry
+      run: |
+        python -m pip install --upgrade pip
+        pip install poetry
+
+    - name: Install dependencies
+      run: |
+        poetry install
+        # Set the `CODEQL-PYTHON` environment variable to the Python executable
+        # that includes the dependencies
+        echo "CODEQL_PYTHON=$(which python)" >> $GITHUB_ENV
```

```suggestion
        echo "CODEQL_PYTHON=$(poetry env info -p)" >> $GITHUB_ENV
```
as by judging:
>         # Set the `CODEQL-PYTHON` environment variable to the Python executable
>         # that includes the dependencies

It looks like we're aiming for the deps installed by Poetry, which usually creates a venv with these isolated from the default system/active `python`. Or maybe I misunderstood the comment, but then I don't understand why do we issue before a `poetry install` command.

---

I somehow thought poetry would install globally here but yeah you are correct, your suggestion should be way more correct.